client/app/components/account-config.coffee

```diff
@@ -771,6 +776,13 @@ AccountConfigMailboxes = React.createClass
 
     _renderMailboxChoice: (labelText, box) ->
         if @state.id? and @state.mailboxes.value isnt ''
+            mailboxesFlat = {}
+            @state.mailboxes.value.map (mailbox) ->
+                id = mailbox.get 'id'
+                mailboxesFlat[id] = {}
```

Comment:
Or may be to the state itself ? This way this part can stay unmutated. 
Why is selectedID copied to each mailbox ?